projects/combined_sequences/combined_sequences.py

```diff
@@ -223,203 +236,97 @@ def runExperiment(args):
-  #       ("TM PredictedActive",
-  #        "Predicted active cells in temporal sequence layer"),
-  #     ],
-  #     basename=exp.name,
-  #     plotDir=os.path.join(os.path.dirname(os.path.realpath(__file__)),
-  #                          "detailed_plots")
-  #   )
-
-  # Compute overall inference statistics
+  # Compute a number of overall inference statistics
   infStats = exp.getInferenceStats()
-  convergencePoint, accuracy = averageConvergencePoint(
-    infStats,"L2 Representation", 30, 40, 1)
+  convergencePoint, accuracy = exp.averageConvergencePoint(
+    "L2 Representation", 30, 40, 1)
 
   predictedActive = numpy.zeros(len(infStats))
   predicted = numpy.zeros(len(infStats))
   predictedActiveL4 = numpy.zeros(len(infStats))
   predictedL4 = numpy.zeros(len(infStats))
   for i,stat in enumerate(infStats):
-    predictedActive[i] = float(sum(stat["TM PredictedActive C0"][2:])) / len(stat["TM PredictedActive C0"][2:])
-    predicted[i] = float(sum(stat["TM NextPredicted C0"][2:])) / len(stat["TM NextPredicted C0"][2:])
+    predictedActive[i] = float(sum(stat["TM PredictedActive C0"][2:])) / len(
+      stat["TM PredictedActive C0"][2:])
+    predicted[i] = float(sum(stat["TM NextPredicted C0"][2:])) / len(
+      stat["TM NextPredicted C0"][2:])
 
-    predictedActiveL4[i] = float(sum(stat["L4 PredictedActive C0"])) / len(stat["L4 PredictedActive C0"])
-    predictedL4[i] = float(sum(stat["L4 Predicted C0"])) / len(stat["L4 Predicted C0"])
+    predictedActiveL4[i] = float(sum(stat["L4 PredictedActive C0"])) / len(
+      stat["L4 PredictedActive C0"])
+    predictedL4[i] = float(sum(stat["L4 Predicted C0"])) / len(
+      stat["L4 Predicted C0"])
 
-  print "# Sequences {} # features {} # columns {} trial # {} network type {}".format(
-    numSequences, numFeatures, numColumns, trialNum, networkType)
-  print "Average convergence point=",convergencePoint,
-  print "Accuracy:", accuracy
-  print
+  print "# Sequences {} # features {} trial # {}\n".format(
+    numSequences, numFeatures, trialNum)
 
-  # Return our convergence point as well as all the parameters and objects
+  # Return a bunch of metrics we will use in plots
+  args.update({"name": exp.name})
   args.update({"objects": sequences.getObjects()})
   args.update({"convergencePoint":convergencePoint})
   args.update({"sensorimotorAccuracyPct": accuracy})
   args.update({"averagePredictions": predicted.mean()})
   args.update({"averagePredictedActive": predictedActive.mean()})
   args.update({"averagePredictionsL4": predictedL4.mean()})
   args.update({"averagePredictedActiveL4": predictedActiveL4.mean()})
-  args.update({"name": exp.name})
   args.update({"statistics": exp.statistics})
 
-  # Can't pickle experiment so can't return it for batch multiprocessing runs.
-  # However this is very useful for debugging when running in a single thread.
-  # if plotInferenceStats:
-  #   args.update({"experiment": exp})
   return args
 
 
 def runExperimentPool(numSequences,
```

Comment:
This function is not being called anymore. Do you still need it?

Reply:
Yes, I will soon add experiments that need it.